### DISCUSSION TITLE
How to set up the seeds correctly?

### DISCUSSION THREAD

**Discussion by SequelONE** [spatie/laravel-translatable#358] [Q&A] [answered]
My seeding file looks like this:

https://github.com/SequelONE/laravel-auth-backpack/blob/main/database/seeders/MenuItemsTableSeeder.php

I have tried different methods of forming an array:

```
[
                'name' => 'About us',
                'title' => [
                    'en' => 'About us',
                    'de' => 'Über uns',
                    'ru' => 'О нас'
                ],
                'type' => 'page_link',
                'link' => NULL,
                'page_id' => 2,
                'parent_id' => NULL,
                'status' => 1,
            ],
```

But when filling into the database, everything looks like this:

```
{
    "en": {
        "de": "Home",
        "en": "Home",
        "ru": "Главная"
    }
}
```

Field type in the JSON table.

How to correctly form a seed file so that all language versions are added to the database correctly, namely like this?

```
{
        "en": "Home",
        "de": "Home",
        "ru": "Главная"
}
```

**Comment by anibalealvarezs** [comment]
Hello.

I do it this way: https://github.com/anibalealvarezs/projectbuilder-package/blob/master/src/Database/Seeders/PbLanguagesSeeder.php

I use manual JSON encoding because this trait works just for models, and I'm using `upsert` which is an Eloquent's [Builder](https://laravel.com/api/9.x/Illuminate/Database/Eloquent/Builder.html#method_upsert) class' method, not an Eloquent [Model](https://laravel.com/api/9.x/Illuminate/Database/Eloquent/Model.html)'s.

If you want to use models, you'll have to [create the instance](https://laravel.com/docs/9.x/eloquent#generating-model-classes) (one for each record to be stored) and use any of the available methods por persisting the data, like [save](https://laravel.com/api/9.x/Illuminate/Database/Eloquent/Model.html#method_save)

**Comment by SequelONE** [comment, marked as answer]
Thanks. That's how it works!

https://github.com/SequelONE/laravel-auth-backpack/blob/main/database/seeders/MenuItemsTableSeeder.php
